Swift 1.2 Changes

batebobo opened the pull request:
@marulya , @StanislavRadkov,

Please review the changes to the style guide, related to Swift 1.2 syntax. There are no significant differences with previous versions of the Swift syntax, except optional binding.

gregheo committed:
Proposed addition on class vs struct

Text is from @hollance's comment in the issue.

gregheo committed:
Struct-scope constants: More Swift-like?

gregheo committed:
Remove translations

ColinEberhardt committed:
Changed "closure" to "closure expression" in a few locations

see #35

Note, Closure Expressions create Closures, hence the two terms can be used interchangeably in many instances. Apple does this in their language reference, e.g. they use "trailing closure syntax" rather than "trailing closure expression syntax".

However, there are a few instances in our style guide where I think 'Closure Expression' makes more sense.

gregheo committed:
Protocol conformance in extensions

gregheo committed:
Merge pull request #66 from gregheo/translations

No Translations

gregheo committed:
Merge pull request #67 from ColinEberhardt/master

Changed "closure" to "closure expression" in a few locations

gregheo committed:
Optional binding, naming and shadowing

gregheo committed:
Show array enumerate() example

gregheo committed:
Merge pull request #51 from gregheo/issue-51

new struct inits over old C functions

gregheo committed:
Merge pull request #58 from gregheo/issue-58

Merge in Matthijs's struct vs classes

gregheo committed:
Merge pull request #64 from gregheo/64-optional-binding

Variable name for Optional Binding

gregheo committed:
Merge pull request #70 from gregheo/70-protocol-conformance

Protocol conformance in extensions

gregheo committed:
Merge pull request #71 from gregheo/array-enumerate-example

Show array enumerate() example

gregheo committed:
Better method naming example + screenshot

gregheo committed:
Merge pull request #76 from gregheo/method-naming

Better method naming example + screenshot

gregheo committed:
Re-indent tip and screenshot

gregheo committed:
Merge pull request #77 from gregheo/benmorrow-indentation

Re-indent tip and screenshot

gregheo committed:
MARK comments for protocol extensions

gregheo committed:
Merge pull request #73 from gregheo/protocol-extensions-mark

Handle MARK properly for extensions.

gregheo committed:
Fix scroll view / table view typo

jawwad committed:
Add guideline for omitting the get clause on read-only variables

gregheo committed:
Merge pull request #83 from jawwad/omit-get-on-read-only-var

Add guideline for omitting the get clause on read-only variables

jawwad committed:
Minor spacing and style related updates

Update NSString to String
Remove extraneous "() -> () in" after closure brace
Add "-" after "// Mark:" which will add a line in the method list dropdown
Added single space after "//"

jawwad committed:
Update existing guideline to emphasize usage of "let" over "var"

- Update language to place more emphasis on using let
- Add "Preferred" and "Not Preferred" examples so that the guideline stands out
- Update a few other examples to use let instead of var

gregheo committed:
Merge pull request #85 from jawwad/misc-minor-updates

Minor spacing and style related updates

jawwad committed:
Add examples for trailing closure syntax and augment existing guideline

Augment guideline to indicate that a trailing closure is not preferred when using a method that contains multiple closures.

gregheo committed:
Combine some rules and examples

gregheo committed:
Credit

gregheo committed:
Merge modified changes about closures

source: jawwad/improve-trailing-closure-guideline

gregheo committed:
Merge branch 'emphasize-let-over-var' of https://github.com/jawwad/swift-style-guide into jawwad-emphasize-let-over-var

gregheo committed:
Remove examples

gregheo committed:
Merge modified 'jawwad/emphasize-let-over-var'

gregheo committed:
Constants are constants; variables are variables

close #74

gregheo committed:
Always name all parameters

gregheo committed:
Enumerations

close #79

gregheo committed:
Simplify the class prefix section

close #82

gregheo committed:
Show multiple optional binding

gregheo committed:
Formatting

gregheo committed:
Type inference examples

winklerrr committed:
Update prose example and add also a new one

The old example didn't correspond to the examples used in the text above

gregheo committed:
Merge pull request #90 from winklerrr/master

Update prose example and add also a new one

sammyd committed:
Add copyright statement

gregheo committed:
Merge pull request #102 from sammyd/master

Add copyright statement

gregheo committed:
Naming example & external parameters

close #92 close #104

gregheo committed:
Better external name, via @larrylegend

gregheo committed:
Short-style Swift 2 statics

gregheo committed:
enumerate() is a protocol method now

gregheo committed:
Merge pull request #115 from raywenderlich/swift2

Swift 2 updates

sammyd committed:
Update the year in the copyright notice

Keeping things current...

Commit:
Merge pull request #130 from raywenderlich/new-year-new-you

Update the year in the copyright notice

Commit:
Update the copyright notice to an Xcode-friendly format

jawwad committed:
Update copyright statement

Added back the asterisks since it looks a bit nicer this way. The copyright statement was previously updated to a format in which the initial spaces would be preserved when the file was re-indented in Xcode, instead of un-indenting as the previous version did. Turns out that what preserves the space is the inclusion of the extra initial asterisk, i.e. "/**" instead of "/*" so added back the asterisks to the intermediate lines.

Commit:
Merge pull request #134 from jawwad/update-copyright-statement

Update copyright statement

Commit:
Adds project settings spacing requirement.

Commit:
Merge pull request #135 from mmorey/master

Adds project settings spacing requirement.

rayfix committed:
println -> print

rayfix committed:
Merge pull request #140 from rayfix/master

println -> print

rayfix committed:
fix missing println -> print fix

rayfix committed:
Merge pull request #141 from rayfix/master

missing println -> print fix

rayfix committed:
Class names -> Type names

rayfix committed:
Merge pull request #142 from rayfix/master

Class names -> Type names

rayfix committed:
move language to naming section

rayfix committed:
Merge pull request #144 from rayfix/master

move language to naming section

bvushkov-rms committed:
Updates according to Swift 1.2 Syntax. Initial version.

batebobo committed:
Remove redundant information

batebobo committed:
Keep the 4-spaces indentation mindset.

batebobo committed:
Add language section as it was accidentally removed

batebobo committed:
Remove swift 2.0 changes

marulya commented on README.markdown:
````diff
@@ -26,12 +30,12 @@ This style guide is based on the raywenderlich.com Style Guide (https://github.c
 
 ```swift
-let MaximumWidgetCount = 100
+private let maximumWidgetCount = 100
````
Please remove **private** from here. This mixes rules: for privates we use the _ notation.

marulya commented on README.markdown:
```diff
@@ -73,58 +77,69 @@ class WidgetContainer {
-class Guideline {
-    func combineWithString(incoming: String, options: Dictionary?) { ... }
-    func upvoteBy(amount: Int) { ... }
+class Counter {
```
We currently use 4 spaces.

marulya commented on README.markdown:
````diff
@@ -73,58 +77,69 @@ class WidgetContainer {
+Use UpperCamelCase for enumeration values:
 
+```swift
+enum Shape {
````
We currently use 4 spaces.

batebobo committed:
Removed conflicting naming rule

marulya commented on README.markdown:
```diff
@@ -197,20 +222,69 @@ The example above demonstrates the following style guidelines:
     let row: Int, column: Int
 
-    init(row: Int,column: Int) {
+  init(row: Int, column: Int) {
```
Please either revert the 2 space indentation here or decrease with 2 spaces the entire code block.

marulya commented on README.markdown:
````diff
@@ -241,13 +315,37 @@ func setAge(age: Int) {
+**Preferred:**
+```swift
+UIView.animateWithDuration(1.0) {
+  self.myView.alpha = 0
````
Please remove **self** from here. Self references are done only when needed as described in the **Use of Self** section.

marulya commented on README.markdown:
````diff
@@ -293,14 +392,41 @@ Use implicitly unwrapped types declared with `!` only for instance variables tha
 
 ```swift
-myOptional?.anotherOne?.optionalView?.setNeedsDisplay()
+self.textContainer?.textLabel?.setNeedsDisplay()
````
Please remove **self** from here. Self references are done only when needed as described in the **Use of Self** section.

marulya commented on README.markdown:
```diff
@@ -293,14 +392,41 @@ Use implicitly unwrapped types declared with `!` only for instance variables tha
+var volume: Double?
+
+if let unwrappedSubview = optionalSubview {
+  if let realVolume = volume {
```
4 spaces

marulya commented on README.markdown:
````diff
@@ -448,11 +573,41 @@ var color = "red"
 var colour = "red"
 ```
 
+
````
Please remove the Copyright Statement section.

batebobo committed:
Update style guide according to recent comments

Removed a couple of spacing issues and also copyright statement and duplicating language section.

batebobo committed:
Remove use of self as per recent comment
